logstash-core/lib/logstash/persisted_queue_config_validator.rb

```diff
@@ -47,8 +47,9 @@ def check(running_pipelines, pipeline_configs)
         max_bytes = config.settings.get("queue.max_bytes").to_i
         page_capacity = config.settings.get("queue.page_capacity").to_i
         pipeline_id = config.settings.get("pipeline.id")
-        queue_path = config.settings.get("path.queue")
-        pq_page_glob = ::File.join(queue_path, pipeline_id, "page.*")
+        queue_path = Paths.get(config.settings.get("path.queue"), pipeline_id).to_s
```

Comment:
If you take a look at the [PR description user created](https://github.com/elastic/logstash/pull/14211), they created a file system for each pipeline and mounted to the `/opt/logstash/queue/pipeline_id`. My assumption is the base path `/opt/logstash/queue` in a reality in different file system. So, giving a room for each pipeline to use its own file system would be an ideal logic.
`queue_path` will include the pipeline ID and when [defining the file system, the logic](https://github.com/mashhurs/logstash/blob/d886c5206717cc62ee5a15a7f265142b28c64130/logstash-core/lib/logstash/persisted_queue_config_validator.rb#L95) retrieves the exact file system rather then the base path volume located.

I still couldn't reproduce the case since my env does not allow to create FS/mount volume (only 1 I was able to) but posted my question to clarify the base paths' file system. Let me know if I am missing anything else.

Thanks.

Reply:
To add more context, linking pipeline path to file system (tested and customer also mentioned below) uglifies the path where `ln` includes additional the `pipeline_id` (customer wants: `data/queue/pipeline_id`, after linking path becomes `data/queue/pipeline_id/pipeline_id`). 

So, IMO our current approach (LS to check pipeline file system) seems a best practise for this case.